app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/ZimFileSelectFragment.kt

```diff
@@ -111,9 +111,15 @@ class ZimFileSelectFragment : BaseFragment() {
     if (savedInstanceState != null && savedInstanceState.getBoolean(WAS_IN_ACTION_MODE)) {
       zimManageViewModel.fileSelectActions.offer(FileSelectActions.RestartActionMode)
     }
+
+    zimManageViewModel.currentPage.observe(viewLifecycleOwner, Observer<Int> { currentPage ->
+      if (currentPage == 1) {
```

Comment:
I'd prefer if this logic was handled outside of the view. LiveData has MoediatorLiveData, had a publishProcessor been used to send the data then there is filter/map and any number of other operators

Reply:
@macgills I am having some trouble in using a publish processor. I've never used it before. Could you please provide some help here?

Reply:
It is even simpler, you can just use `.map` to transform your liveData. You will want to rename some things also as it will no longer be `currentPage` being observed here

Reply:
ehhh after looking at the Transformations API for liveData I like it a lot less.

PublishProcessor would be a lot handier. A publish processor is like a multicast queue

If in a client I have
```
theProcessor.subscribe { print(it) }
```

then every time I call `theProcessor.offer("test")` that sends "test" to every subscription to `theProcessor`

RxJava has a lot of operators that do handy things like so we could say

```
theProcessor.map{ it +"mapped!"}.subscribe{ print(it)}
```

and then we would get ` testmapped!` printed. Map is way more powerful than this 

eg
```
theProcessor.map{ it.equals("test")}. subscribe{ if (it) print("it was test") else print("it wasn't test")}
```
where we can change the type from what was originally sent to the processor.

See how far you can get with a processor and I'll be back tomorrow to review

Reply:
@macgills thanks a lot for the above explanation :)
I have updated the pr using a publish processor. It works and turns off the action mode on changing tabs. Please let me know if the way I have used the publish processor is correct.